tests/shardddl1/run.sh

```diff
@@ -0,0 +1,480 @@
+}
+
+function DM_014_CASE() {
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add column new_col1 int;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values (1,1)"
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add column new_col2 int;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values (2,2,2)"
+    run_sql_source1 "alter table ${shardddl1}.${tb2} add column new_col2 int;"
+    run_sql_source1 "insert into ${shardddl1}.${tb2} values (3,3)"
+    run_sql_source1 "alter table ${shardddl1}.${tb2} add column new_col1 int;"
+    run_sql_source1 "insert into ${shardddl1}.${tb2} values (4,4,4)"
+    run_sql_tidb_with_retry "select count(1) from ${shardddl}.${tb};" "count(1): 4"
+}
+
+function DM_014() {
+    run_case 014 "single-source-optimistic" "init_table 111 112" "clean_table" ""
+}
+
+function DM_015_CASE() {
+    run_sql_source1 "drop database ${shardddl1};"
+    check_log_contain_with_retry "skip event, need handled ddls is empty" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+    run_sql_source1 "create database ${shardddl1};"
+    check_log_contain_with_retry "CREATE DATABASE IF NOT EXISTS \`${shardddl1}\`" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+}
+
+function DM_015() {
+    run_case 015 "single-source-pessimistic" "init_table 111" "clean_table 111" ""
+}
+
+function DM_016_CASE() {
+    run_sql_source1 "drop database ${shardddl1};"
+    check_log_contain_with_retry "skip event, need handled ddls is empty" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+    run_sql_source1 "create database if not exists ${shardddl1};"
+    check_log_contain_with_retry "CREATE DATABASE IF NOT EXISTS \`${shardddl1}\`" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+}
+
+function DM_016() {
+    run_case 016 "single-source-pessimistic" "init_table 111" "clean_table" ""
+}
+
+function DM_017_CASE() {
+    run_sql_source1 "drop table ${shardddl1}.${tb1};"
+    check_log_contain_with_retry "skip event, need handled ddls is empty" $WORK_DIR/worker1/log/dm-worker.log  $WORK_DIR/worker2/log/dm-worker.log
+    run_sql_source1 "create table ${shardddl1}.${tb1}(id int);"
+    check_log_contain_with_retry "CREATE TABLE IF NOT EXISTS \`${shardddl1}\`.\`${tb1}\` (\`id\` INT)" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log
+}
+
+function DM_017() {
+    run_case 017 "single-source-pessimistic" "init_table 111" "clean_table" ""
+}
+
+function DM_018_CASE() {
+    run_sql_source1 "drop table ${shardddl1}.${tb1};"
+    check_log_contain_with_retry "skip event, need handled ddls is empty" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+    run_sql_source1 "create table if not exists ${shardddl1}.${tb1}(id int);"
+    check_log_contain_with_retry "CREATE TABLE IF NOT EXISTS \`${shardddl1}\`.\`${tb1}\` (\`id\` INT)" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log
```

Comment:
create table will success, but here check the same log with case_017, how to distinguish them?

Reply:
we truncate log file every case here 
https://github.com/pingcap/dm/blob/66632e699531475e04a1d6b59fd8c0b9e2571fff/tests/_utils/test_prepare#L121

Reply:
Both cases will success, have updated doc.